### PR TITLE
Support for "left-style" pointers with new --pointer-style option

### PR DESCRIPTION
**Pull request by ethteck**
This PR adds a new runtime flag which allows the user to configure pointers to be next to the type name rather than the variable name. Default behavior is unchanged.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -222,6 +222,13 @@ def parse_flags(flags: List[str]) -> Options:
         action="store_true",
         help="put braces on separate lines",
     )
+    group.add_argument(
+        "--pointer-style",
+        dest="pointer_style",
+        help="control whether to output pointer asterisks next to the type name (left) or next to the symbol name "
+             "(right)",
```
Comment (simonlindholm) [inline src/main.py:229]: `choices=["left", "right"],`

File: src/options.py [hunk 2]
```diff
@@ -9,6 +9,7 @@ class CodingStyle:
     newline_after_function: bool = attr.ib()
     newline_after_if: bool = attr.ib()
     newline_before_else: bool = attr.ib()
+    pointer_style: str = attr.ib()
```
Comment (simonlindholm) [inline src/options.py:12]: Make this a bool (and rename accordingly)

File: src/flow_graph.py [hunk 3]
```diff
@@ -765,18 +765,18 @@ def process(item: Union[Instruction, Label]) -> None:
...
+    dominators: Set["Node"] = attr.ib(init=False, factory=set, repr=False)
+    immediate_dominator: Optional["Node"] = attr.ib(init=False, default=None, repr=False)
+    immediately_dominates: List["Node"] = attr.ib(init=False, factory=list, repr=False)
+    postdominators: Set["Node"] = attr.ib(init=False, factory=set, repr=False)
+    immediate_postdominator: Optional["Node"] = attr.ib(init=False, default=None, repr=False)
+    immediately_postdominates: List["Node"] = attr.ib(init=False, factory=list, repr=False)
     # This is only populated on the head node of the loop,
     # i.e. there is an invariant `(node.loop is None) or (node.loop.head is node)`
-    loop: Optional["NaturalLoop"] = attr.ib(init=False, default=None)
+    loop: Optional["NaturalLoop"] = attr.ib(init=False, default=None, repr=False)
```
Comment (simonlindholm) [inline src/flow_graph.py:779]: Hm, did you experiment with adding a `__repr__` that delegates to `__str__`? This does feel a bit error-prone (and derived classes probably need repr=False too). Alternatively you can do `repr=False` in attr.s

File: src/main.py [hunk 4]
```diff
@@ -313,6 +321,7 @@ def parse_flags(flags: List[str]) -> Options:
         newline_after_function=args.allman,
         newline_after_if=args.allman,
         newline_before_else=args.allman,
+        pointer_style_left=args.pointer_style.lower() == "left",
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/main.py:324]: No clue, just me being paranoid. I'll check

[the remark this answers is not in the document]
Comment (ethteck) [reply src/main.py:324]: Yeah, it is necessary - passing `Left` shows an invalid choice without the lower(). However, we can just say that `Left` isn't valid..what do you think?

[the remark this answers is not in the document]
Comment (simonlindholm) [reply src/main.py:324]: `Left` not being valid sounds good to me. The `lower()` call here surely wouldn't avoid that error though, right? it just fixes `Left` -> `left` *after* validation

[the remark this answers is not in the document]
Comment (ethteck) [reply src/main.py:324]: ahh right..good point :)

**Pull request merged**